src/sync.cpp

```diff
@@ -207,14 +207,27 @@ void AssertLockHeldInternal(const char *pszName, const char *pszFile, int nLine,
 // from the lockorder map
 #ifdef DEBUG_LOCKORDER
 CCriticalSection::CCriticalSection() : name(NULL) {}
-CCriticalSection::CCriticalSection(const char *n) : name(n) {}
+CCriticalSection::CCriticalSection(const char *n) : name(n)
+{
+// print the address of named critical sections so they can be found in the mutrace output
+#ifdef ENABLE_MUTRACE
+    if (name)
+    {
+        printf("CCriticalSection %s at %p\n", name, this);
```

Comment:
Why not saving this log line in the `debug.lo` rather than print to `stdout`?

Reply:
because the mutrace output goes to stdout, so this keeps it all in one place